prow/plugins/releasenote/releasenote.go

```diff
@@ -255,6 +255,11 @@ func handlePR(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent) e
 	var comments []github.IssueComment
 	labelToAdd := determineReleaseNoteLabel(pr.PullRequest.Body, prLabels)
 
+	//Do not handle ReleaseNoteLabelNeeded do not merge label when the PR is merged
+	if labelToAdd == labels.ReleaseNoteLabelNeeded && pr.PullRequest.Merged {
```

Comment:
I think now it should be ok. Analyzed all do-not-merge flags in this relm and it seems so this one is the only that could occur inside the function. So, if flag and merged we should stop.

Reply:
nit: the `labelToAdd == labels.ReleaseNoteLabelNeeded` part is the same like in the `if` below so maybe it would be simpler to add a 
```
if pr.PullRequest.Merged {
  return nil
}
``` 

there but /shrug

Reply:
@petr-muller you are right. I am an if-else master. Corrected now.